xlators/mount/fuse/src/fuse-bridge.c

```diff
@@ -6910,22 +6910,22 @@ init(xlator_t *this_xl)
     /* What you need to find with we do 'df -h' */
     cmd_args = &this_xl->ctx->cmd_args;
     if (cmd_args->fs_display_name) {
-        strcpy(fsname, cmd_args->fs_display_name);
+        strncpy(fsname, cmd_args->fs_display_name, PATH_MAX);
     } else {
         if (cmd_args->volfile_server) {
-            strcpy(fsname, cmd_args->volfile_server);
-            strcat(fsname, ":");
-            strcat(fsname, cmd_args->volfile_id);
+            strncpy(fsname, cmd_args->volfile_server, PATH_MAX);
+            strncat(fsname, ":", PATH_MAX - 1);
```

Comment:
The `size` argument of `strncpy()` and `strncat()` specifies the maximum number of bytes to copy from the source string. For `strncpy()` it might be ok, but for `strncat()` it doesn't prevent an overflow in destination string. To do so we should pass the size of destination string minus the already filled length.

Probably an `snprintf()` will be easier to use.

Reply:
agree, `snprintf()` surely looks better.